x-pack/plugins/alerts/server/saved_objects/migrations.ts

```diff
@@ -12,53 +12,48 @@ import {
 import { RawAlert } from '../types';
 import { EncryptedSavedObjectsPluginSetup } from '../../../encrypted_saved_objects/server';
 
+export const LEGACY_LAST_MODIFIED_VERSION = 'pre-7.10.0';
+
 export function getMigrations(
   encryptedSavedObjects: EncryptedSavedObjectsPluginSetup
 ): SavedObjectMigrationMap {
-  const alertsMigration = changeAlertingConsumer(encryptedSavedObjects, 'alerting', 'alerts');
-
-  const infrastructureMigration = changeAlertingConsumer(
-    encryptedSavedObjects,
-    'metrics',
-    'infrastructure'
-  );
+  const migrationWhenRBACWasIntroduced = markAsLegacyAndChangeConsumer(encryptedSavedObjects);
 
   return {
-    '7.10.0': (doc: SavedObjectUnsanitizedDoc<RawAlert>, context: SavedObjectMigrationContext) => {
-      if (doc.attributes.consumer === 'alerting') {
-        return executeMigration(doc, context, alertsMigration);
-      } else if (doc.attributes.consumer === 'metrics') {
-        return executeMigration(doc, context, infrastructureMigration);
-      }
-      return doc;
-    },
```

Comment:
As we now migrate every single Saved Object as part of this upgrade, it felt cleaner to flatten this out.